src/marlin_stubs/pause/pause.cpp

```diff
@@ -158,163 +193,46 @@ bool load_filament(const float &slow_load_length /*=0*/, const float &fast_load_
     const bool show_lcd /*=false*/, const bool pause_for_user /*=false*/,
     const PauseMode mode /*=PAUSE_MODE_PAUSE_PRINT*/
         DXC_ARGS) {
-    #if !HAS_LCD_MENU
     UNUSED(show_lcd);
-    #endif
 
-    if (!ensure_safe_temperature(mode)) {
-    #if HAS_LCD_MENU
-        if (show_lcd)
-            lcd_pause_show_message(PAUSE_MESSAGE_STATUS, mode);
-    #endif
+    if (!ensure_safe_temperature_notify_progress(PhasesLoadUnload::WaitingTemp, 10, 30)) {
         return false;
     }
 
-    if (pause_for_user) {
-    #if HAS_LCD_MENU
-        if (show_lcd)
-            lcd_pause_show_message(PAUSE_MESSAGE_INSERT, mode);
-    #endif
-        SERIAL_ECHO_MSG(_PMSG(MSG_FILAMENT_CHANGE_INSERT));
-
-    #if HAS_BUZZER
-        filament_change_beep(max_beep_count, true);
-    #else
-        UNUSED(max_beep_count);
-    #endif
-
-        KEEPALIVE_STATE(PAUSED_FOR_USER);
-        wait_for_user = true; // LCD click or M108 will clear this
-    #if ENABLED(HOST_PROMPT_SUPPORT)
-        const char tool = '0'
-        #if NUM_RUNOUT_SENSORS > 1
-            + active_extruder
-        #endif
-            ;
-        host_prompt_reason = PROMPT_USER_CONTINUE;
-        host_action_prompt_end();
-        host_action_prompt_begin(PSTR("Load Filament T"), false);
-        SERIAL_CHAR(tool);
-        SERIAL_EOL();
-        host_action_prompt_button(PSTR("Continue"));
-        host_action_prompt_show();
-    #endif
-    #if ENABLED(EXTENSIBLE_UI)
-        ExtUI::onUserConfirmRequired_P(PSTR("Load Filament"));
-    #endif
-        while (wait_for_user) {
-    #if HAS_BUZZER
-            filament_change_beep(max_beep_count);
-    #endif
-            idle(true);
-        }
-    }
-
-    #if HAS_LCD_MENU
-    if (show_lcd)
-        lcd_pause_show_message(PAUSE_MESSAGE_LOAD, mode);
-    #endif
+    fsm_change(ClinetFSM::Load_unload, GetPhaseIndex(PhasesLoadUnload::UserPush), 30, 0);
+    while (ClientResponseHandler::GetResponseFromPhase(PhasesLoadUnload::UserPush) != Response::Continue)
+        idle(true);
 
-    #if ENABLED(DUAL_X_CARRIAGE)
-    const int8_t saved_ext = active_extruder;
-    const bool saved_ext_dup_mode = extruder_duplication_enabled;
-    active_extruder = DXC_ext;
-    extruder_duplication_enabled = false;
-    #endif
+    //check FILAMET SENSOR
+    //todo
```

Comment:
Ambiguous todo.

Reply:
there is missing implementation of filament sensor state handling
merged 2 comments in one